config.go

```diff
@@ -7,9 +7,89 @@ import (
 	"time"
 )
 
+// These are the versions of the protocol (which includes RPC messages as
+// well as Raft-specific log entries) that this server can _understand_. Use
+// the ProtocolVersion member of the Config object to control the version of
+// the protocol to use when _speaking_ to other servers. This is not currently
+// written into snapshots so they are unversioned. Note that depending on the
```

Comment:
Adding the snapshot versioning uncovered an issue where we have dropped support for old peers in the snapshot (we can read it but not produce it) - we will need to write both in order to upgrade so that old servers can read new snapshots during the migration. I'll unpack this at the same time as snapshot versioning.